src/components/Input/BoolInput.tsx

```diff
@@ -1,35 +1,45 @@
 import { JSX } from 'react';
-import { Checkbox, Group } from '@mantine/core';
+import { Checkbox, Group, MantineStyleProps } from '@mantine/core';
 
-import { Label } from '../Label/Label';
+import { InfoBox } from '../InfoBox/InfoBox';
 
-interface Props {
+interface Props extends MantineStyleProps {
   name: string;
-  description: string | JSX.Element;
-  value: boolean;
+  value?: boolean; // This is optional to allow for defaultChecked to work
   setValue: (value: boolean) => void;
-  label?: JSX.Element;
+  showLabel?: boolean;
+  defaultChecked?: boolean;
+  info?: string | JSX.Element;
   disabled?: boolean;
 }
 
+/**
+ * This custom checkbox should be used for all boolean inputs, to make sure that we use a
+ * consistent design, always support input using Enter, an provide an aria-label for
+ * screen readers.
+ */
 export function BoolInput({
   value,
   setValue,
   name,
-  description,
-  label,
-  disabled = false
+  defaultChecked,
+  info,
+  showLabel = true,
+  disabled = false,
+  ...styleProps
 }: Props) {
   return (
-    <Group gap={'xs'} wrap={'nowrap'}>
+    <Group gap={'xs'} wrap={'nowrap'} {...styleProps}>
       <Checkbox
         checked={value}
         onChange={(event) => setValue(event.currentTarget.checked)}
         onKeyDown={(event) => event.key === 'Enter' && setValue(!value)}
         disabled={disabled}
+        defaultChecked={defaultChecked}
+        label={showLabel ? name : undefined}
         aria-label={name}
```

Comment:
since we use name for aria label, should throw an error if name is set to an empty string? It seems to me that we always expect here that name is set to something

Reply:
Makes sense!